### PR TITLE
GHA to create issue on new IANA release

### LINKED ISSUE #169: GHA workflow to notify on IANA changes?
We're using your fantastic efforts at https://github.com/valhalla/valhalla and recently we added a GHA workflow to auto-create a PR when https://github.com/eggert/tz had a new release. Since your repo is not coupled to that one, would you consider adding a similar workflow? I think it could help with
- less cognitive overhead for you, it could create an issue where you could comment "no need for an update" or "will get to it soon" or so
- more communicative for dependents such as us

We do wait a while before merging such an auto-PR to see if anything changes in this repo, so things are consistent, e.g. here: https://github.com/valhalla/valhalla/pull/4549. A GHA workflow would just be complimentary to your efforts and help a little bit. I'd be happy to script smth together that fits your repo better (e.g. create an issue instead of a PR), but of course no worries if you don't want to include anything like this, it does add a bit more overhead as well:)

### PR DESCRIPTION
**Pull request by nilsnolde**
fixes #169 

not sure how you cut releases, but we could also create a PR which already changes the `TZ_RELEASE` file to reflect the latest release, which you can pull and push?

### CODE REVIEW

**Comment by nilsnolde** [comment]
I tried it on this branch and it created this issue: https://github.com/nilsnolde/timezone-boundary-builder/issues/1

File: .github/workflows/tz_update_check.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Check for new IANA release
+
+on:
+  schedule:
+    - cron: "0 0 * * 0"
+  workflow_dispatch:
```
Comment (nilsnolde) [inline .github/workflows/tz_update_check.yml:6]: not sure how familiar you are with GHA, but this allows you to trigger this action manually in the "Actions" tab

File: .github/workflows/tz_update_check.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Check for new IANA release
+
+on:
+  schedule:
+    - cron: "0 0 * * 0"
```
Comment (nilsnolde) [inline .github/workflows/tz_update_check.yml:5]: and it runs once a week (sunday midnight)

**Review by evansiroky — APPROVED** [review]
Thanks! I'm excited to see it in action.

**Pull request merged**